main.cpp

```diff
@@ -832,11 +832,18 @@ class Framework {
           lv_obj_del(screen_off_label);
         }
       }
+
       if (print_memory_usage) {
-        // print free memory with the knowledge that 14KiB RAM is the actual PineTime-Memory
         lv_mem_monitor(&mem_mon);
-        printf("actual free_size = %d\n", int64_t(mem_mon.free_size) - (LV_MEM_SIZE - 14U*1024U));
+        if (mem_mon.free_size != mem_mon_last_free_size) {
+          uint32_t pinetime_lvgl_memory = 14U*1024U; // 14KiB is the LVGL memory size used in InfiniTime
+          uint32_t mem_used = LV_MEM_SIZE - int64_t(mem_mon.free_size);
```

Comment:
I think this `int64_t` cast isn't needed anymore if we assume, that free_size can't be greater than `LV_MEM_SIZE`. Furthermore in the case `free_size` actually gets greater than `LV_MEM_SIZE` the cast to `int64_t`, then compute the diff, and then cast back to `uint32_t` will still lead to a negative number casted to an unsigned number, resulting in a huge number

i propose to add an `assert(LV_MEM_SIZE >= mem_mon.free_size)` statement to at least check for that case in debug builds. And to remove the `int64_t` cast here

Reply:
I removed the cast, but decided against adding the proposed `assert`, since [LVGL allocates only `LV_MEM_SIZE` bytes](https://github.com/lvgl/lvgl/blob/v7.11.0/src/lv_misc/lv_mem.c#L120) with `LV_MEM_CUSTOM=0` and `LV_MEM_ADR=0`, which InfiniTime uses.

So `lv_mem_monitor` reporting a `free_size` larger than `LV_MEM_SIZE` would be a major bug in LVGL and if we were to guard against that, the `assert` should go higher up so it is not only tripped when memory debugging is toggled :)